src/tabs/tabs.tsx

```diff
@@ -207,16 +209,39 @@ function TabList({
                     width: `${selectedTab.offsetWidth}px`,
                 })
             }
-        }
+        },
+        [selectedId],
+    )
+
+    React.useEffect(
+        function updateSelectedTabPositionOnTabChange() {
+            updateSelectedTabPosition()
```

Comment:
It'd be better to have `updateSelectedTabPosition` accept `selectedId` as a parameter to make the dependency here explicit, and `updateSelectedTabPosition` itself can then have no dependency.

Reply:
> and `updateSelectedTabPosition` itself can then have no dependency.

If `updateSelectedTabPosition` accepts `selectedId` as parameter, every call to `updateSelectedTabPosition` needs to be updateSelectedTabPosition(selectedId)`. This will force every hook making this call to need the `selectedId` dependency, including `observeTabListWidthChange`, which we only want to be called once to initialize the observer.

Reply:
> This will force every hook making this call to need the selectedId dependency, including observeTabListWidthChange, which we only want to be called once to initialize the observer.

This is both true and false 🙂 While `observeTabListWidthChange`'s only explicit dependency is `updateSelectedTabPosition`, `updateSelectedTabPosition` depends on `selectedId` so a new reference would be created when that changes, making `observeTabListWidthChange` also rerun when `selectedId` changes. Extracting that into a param and making that an explicit dependency for consumers would make it clearer that this is happening.